### PR TITLE
[Core] Fixed incorrect intent color for <Icon /> inside <Tree />

### LINKED ISSUE #3042: <Icon />s with an intent in <Tree />s do not have the correct color
#### Environment

- __Package version(s)__: 

    "@blueprintjs/core": "^3.7.0",
    "@blueprintjs/icons": "^3.2.0",

#### Steps to reproduce

1. Create a <Tree /> with one element, where the element icon is an <Icon /> with an intent

#### Actual behavior

It is not the correct color

#### Expected behavior

It should be the correct color

#### Extra info

Repro: https://codesandbox.io/s/mj6nnkkmwx

### PR DESCRIPTION
**Pull request by jacekjagiello**
#### Checklist

- [x] Includes tests
- [x] Update documentation

#### Changes proposed in this pull request:

Fixed not applying intent color for icons inside tree component

#### Reviewers should focus on:
Is there a better solution to fix this?

#### Screenshot
preview

https://68-144397854-gh.circle-artifacts.com/0/home/circleci/project/packages/docs-app/dist/index.html#core/components/tree

### CODE REVIEW

**Comment by adidahiya** [comment]
@jacekjagiello CI didn't run for this commit. Did you by chance enable CircleCI for your fork? Try disabling it and pushing another commit so the build can run through this repo. See the [contributing docs](https://github.com/palantir/blueprint/blob/develop/CONTRIBUTING.md#developing) for more info

**Comment by jacekjagiello** [comment]
@adidahiya yes, I had enabled CiricleCI for my fork, since I think it was required some time ago. Anyway, this and #3354 is now ready to review :)

**Comment by adidahiya** [comment]
docs preview: https://24412-71939872-gh.circle-artifacts.com/0/home/circleci/project/packages/docs-app/dist/index.html#core/components/tree

File: packages/core/src/common/_mixins.scss [hunk 1]
```diff
@@ -25,6 +25,10 @@ $pt-dark-intent-text-colors: (
   "danger" : $red5,
 ) !default;
 
+@mixin color-set-intent($intentName) {
```
Comment (adidahiya) [inline packages/core/src/common/_mixins.scss]: can you rename this to `intent-color`?

Comment (jacekjagiello) [reply packages/core/src/common/_mixins.scss]: @adidahiya done

**Comment by adidahiya** [comment]
thanks @jacekjagiello, this looks good. in the future, please avoid force pushing to this repo, it's easier to review if you only push additional commits. we squash them all at the end of the PR.

**Comment by Cody0913** [comment]
Not sure if relevant or would be handled as well, but I am having the same issue with the MenuItem component. It being overwritten by the menu color.
image

**Comment by adidahiya** [comment]
@Cody0913 can you file a new issue for that please? Ideally with a screenshot of the actual component you're rendering

**Comment by nemosupremo** [comment]
I don't think this is fixed if you are using `bp3-dark`

**Comment by jacekjagiello** [comment]
@Cody0913 @nemosupremo I will check these issues, and try to fix them

**Pull request merged**